Jellyfin.Data.Migrations/JellyfinDbFactory.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Jellyfin.Data.DbContexts;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Design;
+
+namespace Jellyfin.Data.Migrations
+{
+    class JellyfinDbFactory : IDesignTimeDbContextFactory<JellyfinDb>
```

Comment:
If I fix the null return below and run `dotnet ef migration add Foo` in this project directory I get the following error message, I wonder if the cause is that this needs to be in the same project as the DbContext?

> Your target project 'Jellyfin.Data.Migrations' doesn't match your migrations assembly 'Jellyfin.Data'. Either change your target project or change your migrations assembly.
Change your migrations assembly by using DbContextOptionsBuilder. E.g. options.UseSqlServer(connection, b => b.MigrationsAssembly("Jellyfin.Data.Migrations")). By default, the migrations assembly is the assembly containing the DbContext.
Change your target project to the migrations project by using the Package Manager Console's Default project drop-down list, or by executing "dotnet ef" from the directory containing the migrations project.

What was the reason for splitting up the projects this way? Normally I would put this and the migrations together with the dbContext and maybe split out the models into their own netstandard project, so this seems like a strange setup to me

Reply:
So the original reason for splitting it into its own project was because Migrations require the underlying project to target .NET Core, not .NET Standard, but this is still a problem, as that applies to the project that contains the `DbContext` as well. I would like to migrate the projects to target .NET Core, but people in chat seemed wary of the idea.

Reply:
To fix the above error, you would have to change the line `optionsBuilder.UseSqlite("./jellyfin.db");` to `optionsBuilder.UseSqlite("./jellyfin.db", x => x.MigrationsAssembly("Jellyfin.Data.Migrations"));`

Reply:
Ah okay great, as long as there is a way to make the `dotnet ef` tools work I'm fine with splitting them this way.